scripts/release/create-github-draft-release.sh

```diff
@@ -7,51 +7,57 @@ set -euo pipefail
 #   ./scripts/release/create-github-draft-release.sh <tag name>
 #
 
-TAG=$1
-
-# Variables needed for github actions
-GITHUB_TOKEN=$(./scripts/release/get_token.js)
-
-# Set git config
-git config --global user.email "maps_sdk_ios@mapbox.com"
-git config --global user.name "Maps SDK github release bot"
-
-# URL for where we are posting a new release
-URL="https://api.github.com/repos/mapbox/mapbox-maps-ios/releases"
-
-# Custom message for release
-MESSAGE="### Dependency requirements:\n\
-\n\
-* Compatible version of MapboxCoreMaps:\n\
-* Compatible version of MapboxCommon:\n\
-* Compatible version of Xcode:\n\
-* Compatible version of macOS:\n\
-\n\
-### Changes\n\
-\n\
-<Compose changelog here>\n\
-\n\
-### Direct download\n\
-\n\
-Link to download binaries (append your own Mapbox access token [scoped with \`DOWNLOADS:READ\`](https://account.mapbox.com/)):\n\
-\n\
-\`\`\`\n\
-https://api.mapbox.com/downloads/v2/mobile-maps-ios/releases/ios/$TAG/MapboxMaps.zip?access_token=<access-token>\n\
-\`\`\`"
-
-# Body that is passed to the POST request
-BODY="{\"tag_name\":\"v$TAG\",\"target_commitish\":\"main\",\"name\":\"v$TAG\",\"body\":\"$MESSAGE\",\"draft\":true,\"prerelease\":true}"
-
-# Performing the request using github API
-CURL_RESULT=0
-HTTP_CODE=$(curl $URL \
-    -H "Authorization: token $GITHUB_TOKEN" \
-    -H "Accept: application/vnd.github.v3+json" \
-    -d "$BODY" -w "%{response_code}") || CURL_RESULT=$?
-
-if [[ $CURL_RESULT != 0 ]]; then
-    echo "Failed to create draft release (curl error: $CURL_RESULT)"
-    exit $CURL_RESULT
-fi
-
-echo "Result from draft release creation: $HTTP_CODE"
+TAG=${1:-$VERSION}
+
+set -euo pipefail
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+UTILS_PATH="$SCRIPT_DIR/../utils.sh"
+
+# shellcheck source=../utils.sh
+source "$UTILS_PATH"
+
+main() {
+    GITHUB_TOKEN=$(mbx-ci github reader token)
+    export GITHUB_TOKEN
+
+    GL_NATIVE_RELEASE_URL=$(gh release view --repo mapbox/mapbox-gl-native-internal "maps-v$(jq -r .MapboxCoreMaps scripts/release/packager/versions.json)" --json url -q .url)
+    COMMON_RELEASE_URL=$(gh release view --repo mapbox/mapbox-sdk-common "v$(jq -r .MapboxCommon scripts/release/packager/versions.json)" --json url -q .url)
+
+    CHANGELOG=$( ([[ $(command -v parse-changelog) ]] && parse-changelog CHANGELOG.md) || echo "<Compose changelog here>" )
+
+    cat << EOF > notes.txt
+### Dependency requirements:
+
+* Compatible version of MapboxCoreMaps:
+* Compatible version of MapboxCommon:
+* Compatible version of Xcode:
+* Compatible version of macOS:
+
+### Changes
+
+$CHANGELOG
```

Comment:
This will be a good starting point, but there will be manual editing expected as well. CHANGELOG.md is meant to be very succinct, but we want release notes to be more descriptive.

Reply:
Right, we still have to work manually with Release notes. The script purpose is to make changelog collection a bit easier